src/conditional.test.ts

```diff
@@ -106,4 +107,41 @@ describe("typing", () => {
     );
     expectTypeOf(result).toEqualTypeOf<"hello" | 123>();
   });
+
+  // Known issue: using `every(typeGuard)` within `conditional` in `pipe` currently cannot narrow
+  // types correctly
+  // This test is passing like this and if the issue ever gets resolved, we'll find out because it gets
+  // broken. See discussion in https://github.com/remeda/remeda/issues/624 for details
+  it("can only infer types from `every` when type guard is defined separately", () => {
+    const input = [1, 2, 3] as Array<number | string>;
+
+    // incorrect narrowing when using `every` directly
+    const resultBroken = pipe(
+      input,
+      conditional([
+        every(isNumber),
+        (arr) => {
+          expectTypeOf(arr).toEqualTypeOf<Array<number | string>>();
+          return true;
+        },
+      ]),
+    );
+
+    expect(resultBroken).toBe(true);
```

Comment:
This is a runtime check in a typing test. Do we need this? it doesn't test anything that is related to the bug